minisom.py

```diff
@@ -71,39 +71,110 @@ def fast_norm(x):
     return sqrt(dot(x, x.T))
 
 
-def asymptotic_decay(learning_rate, t, max_iter):
-    """Decay function of the learning process.
+def inverse_decay_to_zero(learning_rate, t, max_iter):
+    """Decay function of the learning process that asymptotically
+    approaches zero.
+
     Parameters
     ----------
     learning_rate : float
-        current learning rate.
+        Current learning rate.
 
     t : int
-        current iteration.
+        Current iteration.
 
     max_iter : int
-        maximum number of iterations for the training.
+        Maximum number of iterations for the training.
     """
-    return learning_rate / (1+t/(max_iter/2))
+    C = max_iter / 100.0
+    return learning_rate * C / (C + t)
 
 
-def linear_decay(learning_rate, t, max_iter):
-    return learning_rate * (1 - t/max_iter)
+def linear_decay_to_zero(learning_rate, t, max_iter):
+    """Decay function of the learning process that linearly
+    decreases to zero.
 
+    Parameters
+    ----------
+    learning_rate : float
+        Current learning rate.
 
-def inverse_time_decay(learning_rate, t, max_iter):
-    C = max_iter / 100.0
-    return learning_rate * C / (C+t)
+    t : int
+        Current iteration.
+
+    max_iter : int
+        Maximum number of iterations for the training.
+    """
+    return learning_rate * (1 - t / max_iter)
+
+
+def inverse_decay_to_one(sigma, t, max_iter):
```

Comment:
There's no need anymore for the decay functions to be here. They're now better suited as methods just as the neighborhood functions.